tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -53,6 +53,83 @@ static void check_out_type_matches(Tensor result,
   }
 }
 
+inline Tensor & dispatch_arange(Scalar end, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, end);
+}
+
+inline Tensor dispatch_arange(Scalar end, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, end);
+}
+
+inline Tensor & dispatch_arange(Scalar start, Scalar end, Scalar step, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, start, end, step);
+}
+
+inline Tensor dispatch_arange(Scalar start, Scalar end, Scalar step, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, start, end, step);
+}
+
+static inline bool allIntegral(std::initializer_list<Scalar> l) {
+  for (auto s : l) {
+    if (!s.isIntegral() && !(s.isBackedByTensor() && at::isIntegralType(s.toTensor().type().scalarType()))) {
```

Comment:
nit: `!(s.isIntegral() || (s.isBackedByTensor() && at::isIntegralType(...)))` would be more readable I think. This is a bit weird, because you pushed the outer negation one level below, but not all the way down.